### PR TITLE
fix: Kafka batch consumer issues

### PR DESCRIPTION
**Pull request by xWiiLLz**
This PR fixes a problem our batch consumer encountered when the broken coordinating a given consumer group was restarted; the batch consumer would then show up as being alive but keep lagging forever (never consuming new messages).

This PR also makes the switch from Shopify/sarama to IBM/sarama.

### CODE REVIEW

File: pubsubx/kafka.go [hunk 1]
```diff
@@ -24,7 +23,7 @@ var _ Publisher = (*kafkaPublisher)(nil)
 func setupKafkaPublisher(l *logrusx.Logger, c *Config, opts *pubSubOptions) (Publisher, error) {
 	conf := kafkax.PublisherConfig{
 		Brokers:   c.Providers.Kafka.Brokers,
-		Marshaler: kafka.DefaultMarshaler{},
+		Marshaler: kafkax.DefaultMarshaler{},
```
Comment (Sansinj) [inline pubsubx/kafka.go:26]: good catch

File: pubsubx/kafkax/batch_message_handler.go [hunk 2]
```diff
@@ -30,9 +31,36 @@ type batchedMessageHandler struct {
 	messageParser messageParser
 	messages      chan *messageHolder
 	wg            sync.WaitGroup
+	mu            sync.Mutex
 	cancel        context.CancelFunc
 }
 
+// Cleanup implements MessageHandler.
+func (h *batchedMessageHandler) Cleanup(*sarama.ConsumerGroupSession) error {
```
Comment (Sansinj) [inline pubsubx/kafkax/batch_message_handler.go:39]: Am I missing something, I would expect the session to be used in the `Cleanup` and `Setup` method

**Review by Sansinj — APPROVED** [review]

File: pubsubx/kafkax/batch_message_handler.go (hunk 2 above)
Comment (xWiiLLz) [reply pubsubx/kafkax/batch_message_handler.go:39]: No we don't need it for the current implementation as the offsets are committed in order code paths already. The problem we had was really that we were starting the `startProcessing` goroutine in the constructor, and closing it when we stopped processing messages once; but then never restarting that loop after the ProcessMessages was called again (in the case of brokers restart, this is what was happening)

Comment (Sansinj) [reply pubsubx/kafkax/batch_message_handler.go:39]: Yes sorry point was we could do Cleanup() everywhere 😅

**Pull request merged**